tests/runtime/test_command_buffer_multi_device.c

```diff
@@ -120,7 +120,7 @@ main (int _argc, char **_argv)
             devices[j], CL_DEVICE_COMMAND_BUFFER_SYNC_DEVICES_KHR,
             sizeof (cl_device_id) * num_sync_devices, sync_devices, NULL));
 
-          for (unsigned i = 0; i < num_devices; ++i)
+          for (i = 0; i < num_devices; ++i)
```

Comment:
By glance, this suggests that `i` could be used outside the loop but it's not the case so it looks a bit confusing to me. Perhaps, all loops should be changed from `for(i = ...)` to `for (<type> i = ...)`. 

```suggestion
          for (unsigned i = 0; i < num_devices; ++i)
```

Reply:
The function uses the pattern of declaring `i` at the top, which you will often see in older C code or code with goto statements. putting the unsigned back will result in a warning that `i` is being shadowed.

Reply:
IIUC, the top-level `i`can be removed (after all the loops are transformed to to form in the above).

Reply:
I've implemented the suggestiong